uitools/cpp/Esri/ArcGISRuntime/Toolkit/SearchViewController.cpp

```diff
@@ -168,7 +172,24 @@ namespace Toolkit {
 
           if (mapView->isNavigating())
           {
-            setIsEligableForRequery(true);
+            // Check extent difference.
+            double widthDiff = abs(queryArea().extent().width() - lastSearchArea().extent().width());
+            double heightDiff = abs(queryArea().extent().width() - lastSearchArea().extent().width());
+
+            double widthThreshold = lastSearchArea().extent().width() * m_thresholdRatioRepeatSearch;
+            double heightThreshold = lastSearchArea().extent().height() * m_thresholdRatioRepeatSearch;
+            if (widthDiff > widthThreshold || heightDiff > heightThreshold)
+            {
+              setIsEligableForRequery(true);
+            }
+            // Check center difference.
+            double centerDiff = ArcGISRuntime::GeometryEngine::distance(lastSearchArea().extent().center(), queryArea().extent().center());
+            double currentExtentAvg = (lastSearchArea().extent().width() + lastSearchArea().extent().height()) / 2;
+            double threshold = currentExtentAvg * m_thresholdRatioRepeatSearch;
+            if (centerDiff > threshold)
```

Comment:
❓ Do we need this check along with the extent test? Seems redundant after the extent calculations have already been done.

Reply:
it seems so, because the extent one will only check how wide or high (in km for example) is currently shown on the screen. Will trigger an update only for zooming out and in (changes the km being shown on the screen).
The center check is to check if the user pans around but keeping the same zoom

Reply:
Ah that makes sense. If you could put in seperate functions or give them comments denoting that one is a pan test and one is a zoom test that'd be a great help in readibility.